### PR TITLE
Close #22 : Add Continuous Integration

### LINKED ISSUE #22: Add Continuous Integration
We should probably add some Continuous Integration (CI) via http://travis-ci.org/ to automatically run a large sample of tests for the API for each PR before we merge it.

Idea: [pytest](http://pytest.org) and/or [tox](https://tox.readthedocs.org/en/latest/) (shamelessly stolen from [Open Sourcing a Python Project the Right Way](http://www.jeffknupp.com/blog/2013/08/16/open-sourcing-a-python-project-the-right-way/))

### PR DESCRIPTION
**Pull request by RemiLehe**
**Please merge pull request #32 first**

## Overview

Following the advice of @ax3l, this pull requests:
- Creates a set of automated tests. These tests can now be run by typing `python setup.py test`, as mentioned in CONTRIBUTING.md.
- Links them to [Travis CI](https://travis-ci.org/), so that these tests are automatically run every time someone pushes commits to the Github repository. (The tests run on remote servers that provided for free by Travis CI.)

I heavily relied on [Open Sourcing a Python Project the Right Way](https://www.jeffknupp.com/blog/2013/08/16/open-sourcing-a-python-project-the-right-way/), when doing these changes.

NB: I also updated some of the installation instructions as they were no longer valid.

## More details on the automated tests

For the moment they are quite basic: they simply make sure that the notebooks in the `tutorials` directory run without any problem. This is done by converting the notebooks to pure-Python scripts and running these scripts. Therefore, the features that are specific to the IPython notebook (mainly the GUI) are not tested (I did not find a way to test them, but it would be nice to have one !).

## More details on the tests on Travis CI

The Travis CI instructions are given in `.travis.yml`. The tests include:
- Installing openPMD-viewer from scratch, by first installing a light version of Anaconda ([Miniconda](http://conda.pydata.org/miniconda.html)), installing all the listed dependencies of openPMD-viewer, and installing openPMD-viewer itself.
- Running `python setup.py test` to check the tutorial notebooks.

This is done for each push and also each pull requests, resulting in e.g. a nice button below saying that "All checks have passed".

I have set up Travis CI for my own fork of openPMD-viewer, and for the main repository. But I think that for the other forks, their owner have to register to Travis CI themselves if they want the tests to be run on their forks. See [Getting started - Travis CI](https://docs.travis-ci.com/user/getting-started/)

### CODE REVIEW

**Comment by ax3l** [comment]
that is a great update, thank you!

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,6 @@
 *.DS_Store
 build/
 dist/
-opmd_viewer.egg-info/
\ No newline at end of file
+opmd_viewer.egg-info/
+.cache/
+.eggs/
```
Comment (ax3l) [inline .gitignore]: if you want, adding an empty newline in all text/code files makes `diff`s better readable

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  - ./miniconda.sh -b
+  - export PATH=/home/travis/miniconda2/bin:$PATH
+
+# Install packages
+install:
+  - conda install --yes python=$TRAVIS_PYTHON_VERSION --file requirements.txt
+  - conda install --yes python=$TRAVIS_PYTHON_VERSION jupyter
+  - python setup.py install
+script: 
+  - "python setup.py test"
```
Comment (ax3l) [inline .travis.yml]: if you want, adding an empty newline in all text/code files makes `diff`s better readable

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Borrowed from here:  https://gist.github.com/dan-blanchard/7045057
+# The validity of this file can be checked here: http://lint.travis-ci.org/
+
+language: python
+python:
+    - "2.7"
```
Comment (ax3l) [inline .travis.yml]: can we add a `3.X` version, too? e.g., `  - "3.5"`?

**Comment by soerenjalas** [comment]
Wow, I see you've been busy over the holidays :) Sorry for reviewing this just now. Great update!

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[aliases]
+test=pytest
```
Comment (ax3l) [inline setup.cfg]: shall we add an end of file newline? :)

**Comment by ax3l** [comment]
@soerenjalas I just added review comments 5 min ago that you over-run by merging ... ;)

**Comment by ax3l** [comment]
I will add a follow-up PR...

**Comment by soerenjalas** [comment]
@ax3l Sorry for that… Didn't see your comments.

File: tests/test_tutorials.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    for notebook_name in tutorial_notebooks:
+
+        # Do a first pass where only the non-IPython features are tested.
+        # (This gives better debugging information.)
+        # The notebook is converted to a standard Python script and
+        # run directly with `execfile`
+        script_name = notebook_name[:-6] + '.py'
+        os.system( 'ipython nbconvert --to=python %s' %notebook_name )
+        clean_ipython_features( script_name )
+        execfile( script_name )
```
Comment (ax3l) [inline tests/test_tutorials.py]: do you really want to *source* the other script into the current one? Or just execute an other script?
In the latter case, execfile is not the right choice.

**Comment by ax3l** [comment]
@soerenjalas no prob! :)

**Pull request merged**